docs/guides/software-lifecycle/metrics/metrics-starter-kit/README.md

```diff
@@ -0,0 +1,79 @@
+# Apache DevLake Installation Guide
+
+<p align="center">A step-by-step guide for setting up and configuring Apache DevLake using the Starter Kit.</p>
+
+## Introduction
+
+**Welcome to Apache DevLake!** This comprehensive Starter Kit is designed to simplify the installation process and configuration of Apache DevLake, providing you with a smooth and efficient development experience. Whether you are a seasoned developer or just starting with Apache DevLake, this guide and kit aim to enhance your workflow.
+
+**Use Cases**:
+
+- Streamlining Apache DevLake setup and configuration
+- Implementing best practices in Apache DevLake projects
+- Providing comprehensive documentation for Apache DevLake developers
+
+---
+
+## Prerequisites
+
+Before you begin, make sure you have the following installed:
+
+- [Apache DevLake](https://github.com/apache/incubator-devlake)
+- [Docker](https://docs.docker.com/engine/install/)
+
+---
+
+## Quick Start
+
+Ready to dive in? Follow these steps to get started quickly:
+
+**Run the Setup Script:**
+
+-- TODO -- Add proper link to script without commit hash
+
+```bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/NASA-AMMOS/slim/b7a5f795b3ea29809072b3cf1e65e8218db5d73b/docs/guides/software-lifecycle/metrics/metrics-starter-kit/starter-kit/install_devlake.sh"
```

Comment:
```suggestion
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/NASA-AMMOS/slim/b7a5f795b3ea29809072b3cf1e65e8218db5d73b/docs/guides/software-lifecycle/metrics/metrics-starter-kit/starter-kit/install_devlake.sh)"
```

Also - I'm getting an error running this:

```
$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/NASA-AMMOS/slim/b7a5f795b3ea29809072b3cf1e65e8218db5d73b/docs/guides/software-lifecycle/metrics/metrics-starter-kit/starter-kit/install_devlake.sh)"
curl: (22) The requested URL returned error: 404
```

Reply:
Can you try again? I have updated the link

Reply:
Hey @ddalton-jpl - thanks for looking into this! So I tried the following but received exceptions (see below). Also, you need a `)` closing parenthesis at the end of your curl command.

```
$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/ddalton-jpl/slim/issue-117/docs/guides/software-lifecycle/metrics/metrics-starter-kit/install_devlake.sh)"
=== Apache DevLake Installation ===
Step 1: Prerequisites
Make sure you have Docker v19.03.10+ and docker-compose v2.2.3+ installed.
If you have Docker Desktop installed, docker-compose is already included.
Step 2. Downloading docker-compose.yml and env.example
--2024-01-30 15:34:16--  https://github.com/apache/incubator-devlake/releases/download/v0.20.0-beta4/docker-compose.yml
Resolving github.com (github.com)... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/384111310/4bc97aef-8204-441d-ae3f-94e4b797a082?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240130%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240130T233416Z&X-Amz-Expires=300&X-Amz-Signature=365931dfb6ec54b018c615e4bfe7885a1951eab608648b0ad059710556c13ba2&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=384111310&response-content-disposition=attachment%3B%20filename%3Ddocker-compose.yml&response-content-type=application%2Foctet-stream [following]
--2024-01-30 15:34:16--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/384111310/4bc97aef-8204-441d-ae3f-94e4b797a082?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240130%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240130T233416Z&X-Amz-Expires=300&X-Amz-Signature=365931dfb6ec54b018c615e4bfe7885a1951eab608648b0ad059710556c13ba2&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=384111310&response-content-disposition=attachment%3B%20filename%3Ddocker-compose.yml&response-content-type=application%2Foctet-stream
Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.109.133, 185.199.111.133, 185.199.108.133, ...
Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2319 (2.3K) [application/octet-stream]
Saving to: 'docker-compose.yml'

docker-compose.yml             100%[=================================================>]   2.26K  --.-KB/s    in 0s      

2024-01-30 15:34:17 (26.6 MB/s) - 'docker-compose.yml' saved [2319/2319]

--2024-01-30 15:34:17--  https://github.com/apache/incubator-devlake/releases/download/v0.20.0-beta4/env.example
Resolving github.com (github.com)... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/384111310/615b5bca-736c-494d-bd76-82c406122340?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240130%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240130T233418Z&X-Amz-Expires=300&X-Amz-Signature=b263aeefbc2a923ecabe2f48f7966a2e8d19078daa1780c820acee4bbdae51bb&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=384111310&response-content-disposition=attachment%3B%20filename%3Denv.example&response-content-type=application%2Foctet-stream [following]
--2024-01-30 15:34:18--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/384111310/615b5bca-736c-494d-bd76-82c406122340?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240130%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240130T233418Z&X-Amz-Expires=300&X-Amz-Signature=b263aeefbc2a923ecabe2f48f7966a2e8d19078daa1780c820acee4bbdae51bb&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=384111310&response-content-disposition=attachment%3B%20filename%3Denv.example&response-content-type=application%2Foctet-stream
Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.109.133, 185.199.111.133, 185.199.108.133, ...
Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 941 [application/octet-stream]
Saving to: 'env.example'

env.example                    100%[=================================================>]     941  --.-KB/s    in 0s      

2024-01-30 15:34:18 (37.4 MB/s) - 'env.example' saved [941/941]

Step 3. Renaming env.example to .env...
Step 4. Generating encryption key...
Method 2: Setting in .env file
Step 5: Launching DevLake with Docker Compose
Traceback (most recent call last):
  File "urllib3/connectionpool.py", line 670, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1255, in request
  File "http/client.py", line 1301, in _send_request
  File "http/client.py", line 1250, in endheaders
  File "http/client.py", line 1010, in _send_output
  File "http/client.py", line 950, in send
  File "docker/transport/unixconn.py", line 43, in connect
FileNotFoundError: [Errno 2] No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests/adapters.py", line 439, in send
  File "urllib3/connectionpool.py", line 726, in urlopen
  File "urllib3/util/retry.py", line 410, in increment
  File "urllib3/packages/six.py", line 734, in reraise
  File "urllib3/connectionpool.py", line 670, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1255, in request
  File "http/client.py", line 1301, in _send_request
  File "http/client.py", line 1250, in endheaders
  File "http/client.py", line 1010, in _send_output
  File "http/client.py", line 950, in send
  File "docker/transport/unixconn.py", line 43, in connect
urllib3.exceptions.ProtocolError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker/api/client.py", line 214, in _retrieve_server_version
  File "docker/api/daemon.py", line 181, in version
  File "docker/utils/decorators.py", line 46, in inner
  File "docker/api/client.py", line 237, in _get
  File "requests/sessions.py", line 543, in get
  File "requests/sessions.py", line 530, in request
  File "requests/sessions.py", line 643, in send
  File "requests/adapters.py", line 498, in send
requests.exceptions.ConnectionError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 81, in main
  File "compose/cli/main.py", line 200, in perform_command
  File "compose/cli/command.py", line 60, in project_from_options
  File "compose/cli/command.py", line 152, in get_project
  File "compose/cli/docker_client.py", line 41, in get_client
  File "compose/cli/docker_client.py", line 170, in docker_client
  File "docker/api/client.py", line 197, in __init__
  File "docker/api/client.py", line 221, in _retrieve_server_version
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
[30772] Failed to execute script docker-compose
Traceback (most recent call last):
  File "urllib3/connectionpool.py", line 670, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1255, in request
  File "http/client.py", line 1301, in _send_request
  File "http/client.py", line 1250, in endheaders
  File "http/client.py", line 1010, in _send_output
  File "http/client.py", line 950, in send
  File "docker/transport/unixconn.py", line 43, in connect
FileNotFoundError: [Errno 2] No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests/adapters.py", line 439, in send
  File "urllib3/connectionpool.py", line 726, in urlopen
  File "urllib3/util/retry.py", line 410, in increment
  File "urllib3/packages/six.py", line 734, in reraise
  File "urllib3/connectionpool.py", line 670, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1255, in request
  File "http/client.py", line 1301, in _send_request
  File "http/client.py", line 1250, in endheaders
  File "http/client.py", line 1010, in _send_output
  File "http/client.py", line 950, in send
  File "docker/transport/unixconn.py", line 43, in connect
urllib3.exceptions.ProtocolError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker/api/client.py", line 214, in _retrieve_server_version
  File "docker/api/daemon.py", line 181, in version
  File "docker/utils/decorators.py", line 46, in inner
  File "docker/api/client.py", line 237, in _get
  File "requests/sessions.py", line 543, in get
  File "requests/sessions.py", line 530, in request
  File "requests/sessions.py", line 643, in send
  File "requests/adapters.py", line 498, in send
requests.exceptions.ConnectionError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 81, in main
  File "compose/cli/main.py", line 200, in perform_command
  File "compose/cli/command.py", line 60, in project_from_options
  File "compose/cli/command.py", line 152, in get_project
  File "compose/cli/docker_client.py", line 41, in get_client
  File "compose/cli/docker_client.py", line 170, in docker_client
  File "docker/api/client.py", line 197, in __init__
  File "docker/api/client.py", line 221, in _retrieve_server_version
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
[30788] Failed to execute script docker-compose
Step 6: Collect data and view dashboards
Visit http://localhost:4000 in your browser to configure DevLake and collect data.
=== Installation Completed ===
```